include/deal.II/matrix_free/matrix_free.templates.h

```diff
@@ -85,12 +85,17 @@ std::pair<unsigned int, unsigned int>
 MatrixFree<dim, Number, VectorizedArrayType>::create_cell_subrange_hp_by_index(
   const std::pair<unsigned int, unsigned int> &range,
   const unsigned int                           fe_index,
-  const unsigned int                           vector_component) const
+  const unsigned int                           dof_handler_index) const
 {
-  AssertIndexRange(fe_index, dof_info[vector_component].max_fe_index);
+  if (dof_info[dof_handler_index].max_fe_index == 0)
+    return range;
+
+  AssertIndexRange(fe_index, dof_info[dof_handler_index].max_fe_index);
   const std::vector<unsigned int> &fe_indices =
-    dof_info[vector_component].cell_active_fe_index;
-  if (fe_indices.empty() == true)
+    dof_info[dof_handler_index].cell_active_fe_index;
+
+  if (fe_indices.empty() == true ||
+      dof_handlers[dof_handler_index]->get_fe_collection().size() == 1)
```

Comment:
Please not that had to make following modifications to successfully run all `tests/matrix_free`: 
1. if indices are set up (needed by matrix_free/no_index_initialize.debug)
2. in the case of cell catergorization `fe_indices.empty() == true` is not equivalent for the check if hp-capability is enabled or not (needed for `matrix_free/cell_categorization.mpirun=1.debug`)

Reply:
OK.